astropy/io/fits/hdu/image.py

```diff
@@ -928,8 +928,8 @@ def __init__(self, data=None, header=None, do_not_scale_image_data=False,
     def match_header(cls, header):
         card = header.cards[0]
         return (card.keyword == 'SIMPLE' and
-                ('GROUPS' not in header or header['GROUPS'] != True) and
-                card.value == True)
+                ('GROUPS' not in header or header['GROUPS'] is not True) and
```

Comment:
This was tricky, this one caused the regression in #5807, header['GROUPS'] is not a bool, and in the example case it's 2, so the second half of the expression becomes False rather than True